### ISSUE TITLE
Gigantic log files and high CPU/RAM usage, unresponsiveness

### ISSUE THREAD

**Issue by HorstBort** [nextcloud/desktop#2854]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Expected behaviour
Nextcloud should happily run in the background, respond to interactions and not create ginormous logfiles.

### Actual behaviour
After a few minutes, the client becomes unresponsive, menu does appear minutes after clicking the tray icon, settings do not open. CPU usage for nextcloud is at 100%, occupying multiple GBs of RAM, creating multiple GBs of log files in a few minutes.
Granted, I have around 200 000 files in my sync folder.

### Steps to reproduce
1. Install nextcloud client
2. Start with desktop shortcut or via `nextcloud` on cli (**without** `--logdebug`!)
2. Sync folder with ~200 000 files
3. Try to open menu or settings

### Client configuration
    Client version: nextcloud-desktop:
      Installed: 3.1.1-20201225.075704.75c329f4f-1.0~focal1

Operating system: 

    lsb_release -a
    No LSB modules are available.
    Distributor ID: Ubuntu
    Description:    Ubuntu 20.04.1 LTS
    Release:        20.04
    Codename:       focal

OS language: English

Qt version used by client package (Linux only, see also Settings dialog):

    apt-cache policy libqt5gui5
    libqt5gui5:
      Installed: 5.12.8+dfsg-0ubuntu1
      Candidate: 5.12.8+dfsg-0ubuntu1
      Version table:
     *** 5.12.8+dfsg-0ubuntu1 500
            500 http://de.archive.ubuntu.com/ubuntu focal/universe amd64 Packages
            100 /var/lib/dpkg/status

Client package (From Nextcloud or distro) (Linux only): nextcloud-desktop

Installation path of client:

### Server configuration

Nextcloud version: 20.0.4

Storage backend (external storage): - 

### Logs

Please use Gist (https://gist.github.com/) or a similar code paster for longer
logs.

1. Client logfile: 

    Excerpts, b/c 1.5 G in 3.5 minutes:

        head 20210121_1646_owncloud.log.0
         2021-01-21 16:46:57:428 [ info nextcloud.gui.application ]:    "################## Nextcloud locale:[en_GB] ui_lang:[] version:[3.1.1-20201225.075704.75c329f4f-1.0~focal1] os:[Ubuntu 20.04.1 LTS]"
         2021-01-21 16:46:57:428 [ info nextcloud.gui.application ]:    Using "en_GB" translation
         2021-01-21 16:46:57:428 [ info nextcloud.gui.socketapi ]:      server started, listening at  "/run/user/1000/Nextcloud/socket"
         2021-01-21 16:46:57:429 [ info nextcloud.gui.folder.manager ]: setting remote poll timer interval to 30000 msec
         2021-01-21 16:46:57:484 [ info nextcloud.gui.account.manager ]:        Account for QUrl("https://home.muentinga.org/nextcloud") using auth type "webflow"
         2021-01-21 16:46:57:484 [ info nextcloud.sync.credentials.webflow ]:   Get QNAM
         2021-01-21 16:46:57:484 [ info nextcloud.gui.account.manager ]:        Account for QUrl("https://sirmo.intra.dlr.de/nextcloud") using auth type "webflow"
         2021-01-21 16:46:57:484 [ info nextcloud.sync.credentials.webflow ]:   Get QNAM
         2021-01-21 16:46:57:485 [ debug nextcloud.gui.activity ]       [ OCC::User::setNotificationRefreshInterval ]:  Starting Notification refresh timer with  300  sec interval
         2021-01-21 16:46:57:485 [ debug nextcloud.gui.activity ]       [ OCC::User::setNotificationRefreshInterval ]:  Starting Notification refresh timer with  300  sec interval

        tail 20210121_1646_owncloud.log.0
         2021-01-21 16:50:21:498 [ debug nextcloud.sync.database.sql ]  [ OCC::SqlQuery::exec ]:        SQL exec "SELECT path FROM selectivesync WHERE type=?1"
         2021-01-21 16:50:21:498 [ warning nextcloud.sync.engine ]:     ERROR during  csync_update :  "Aborted by the user Aborted by the user"
         2021-01-21 16:50:21:498 [ info nextcloud.sync.engine ]:        Update phase was aborted by user!
         2021-01-21 16:50:21:770 [ info nextcloud.sync.database ]:      Closing DB "/home/muen_hk/Documents/._sync_d008eaab61aa.db"
         2021-01-21 16:50:21:777 [ info nextcloud.sync.engine ]:        CSync run took  192077 ms
         2021-01-21 16:50:21:780 [ info nextcloud.gui.folder ]: Client version 3.1.1-20201225.075704.75c329f4f-1.0~focal1  Qt 5.12.8  SSL  OpenSSL 1.1.1f  31 Mar 2020
         2021-01-21 16:50:21:780 [ info nextcloud.gui.folder ]: SyncEngine finished without problem.
         2021-01-21 16:50:21:780 [ info nextcloud.gui.folder ]: Folder sync result:  1
         2021-01-21 16:50:21:780 [ debug nextcloud.gui.folder ] [ OCC::Folder::slotSyncFinished ]:      Sync failed, keeping last sync's local discovery path list
         2021-01-21 16:50:21:780 [ info nextcloud.gui.application ]:    Sync state changed for folder  "https://sirmo.intra.dlr.de/nextcloud/remote.php/dav/files/hauke.muentinga/" :  "Success"

        wc 20210121_1646_owncloud.log.0
           5718800  128889079 1581626141 20210121_1646_owncloud.log.0

        grep '\[ info' 20210121_1646_owncloud.log.0 | wc
           1556565 22999844 342062531

        grep '\[ debug' 20210121_1646_owncloud.log.0 | wc
           4162193 105888661 1239558041
    Again, this is **without** starting with `--logdebug`!

    A few minutes later: Please note that these files live in a folder **excluded** from sync.

        2021-01-21 17:30:57:934 [ info nextcloud.sync.csync.reconciler ]:       INSTRUCTION_NEW                client dir:  03_Software/Sytems Engineering/doorstop-ng-gui/assets/material-design-icons/android/image/switch_video/materialiconsround/black/res/drawable-xxxhdpi
        2021-01-21 17:30:57:934 [ info nextcloud.sync.csync.reconciler ]:       INSTRUCTION_NEW                client file: 03_Software/Sytems Engineering/doorstop-ng-gui/assets/material-design-icons/ios/file/cloud_done/materialiconsround/black/round_cloud_done_black_36pt.xcassets/round_cloud_done_black_36pt.imageset/Contents.json
        2021-01-21 17:30:57:934 [ info nextcloud.sync.csync.reconciler ]:       INSTRUCTION_NEW                client dir:  03_Software/Sytems Engineering/doorstop-ng-gui/assets/material-design-icons/android/places/fitness_center/materialiconsround/black/res/drawable-mdpi
        2021-01-21 17:30:57:934 [ info nextcloud.sync.csync.reconciler ]:       INSTRUCTION_NEW                client file: 03_Software/Sytems Engineering/doorstop-ng-gui/assets/material-design-icons/ios/file/cloud_done/materialiconsround/black/round_cloud_done_black_36pt.xcassets/round_cloud_done_black_36pt.imageset/round_cloud_done_black_36pt_3x.png
        2021-01-21 17:30:57:934 [ info nextcloud.sync.csync.reconciler ]:       INSTRUCTION_NEW                client dir:  03_Software/Sytems Engineering/doorstop-ng-gui/assets/material-design-icons/ios/file/cloud_done/materialiconsround/black/round_cloud_done_black_36pt.xcassets/round_cloud_done_black_36pt.imageset
        2021-01-21 17:30:57:934 [ info nextcloud.sync.csync.reconciler ]:       INSTRUCTION_NEW                client dir:  03_Software/Sytems Engineering/doorstop-ng-gui/assets/material-design-icons/png/action/settings_input_component/materialiconssharp/36dp 
        2021-01-21 17:30:57:934 [ info nextcloud.sync.csync.reconciler ]:       INSTRUCTION_NEW                client file: 03_Software/Sytems Engineering/doorstop-ng-gui/assets/material-design-icons/android/places/dry/materialiconsround/black/res/drawable/round_dry_24.xml
        2021-01-21 17:30:57:934 [ info nextcloud.sync.csync.reconciler ]:       INSTRUCTION_NEW                client dir:  03_Software/Sytems Engineering/doorstop-ng-gui/assets/material-design-icons/android/image/wb_sunny/materialiconstwotone/black
        2021-01-21 17:30:57:934 [ info nextcloud.sync.csync.reconciler ]:       INSTRUCTION_NEW                client dir:  03_Software/Sytems Engineering/doorstop-ng-gui/assets/material-design-icons/ios/file/cloud_done/materialiconsround/black/round_cloud_done_black_36pt.xcassets
        2021-01-21 17:30:57:934 [ info nextcloud.sync.csync.reconciler ]:       INSTRUCTION_NEW                client file: 03_Software/Sytems Engineering/doorstop-ng-gui/assets/material-design-icons/ios/file/cloud_done/materialiconsround/black/round_cloud_done_black_48pt.xcassets/round_cloud_done_black_48pt.imageset/Contents.json
        2021-01-21 17:30:57:934 [ info nextcloud.sync.csync.reconciler ]:       INSTRUCTION_NEW                client file: 03_Software/Sytems Engineering/doorstop-ng-gui/assets/material-design-icons/ios/file/cloud_done/materialiconsround/black/round_cloud_done_black_48pt.xcassets/round_cloud_done_black_48pt.imageset/round_cloud_done_black_48pt_3x.png
        2021-01-21 17:30:57:934 [ info nextcloud.sync.csync.reconciler ]:       INSTRUCTION_NEW                client file: 03_Software/Sytems Engineering/doorstop-ng-gui/assets/material-design-icons/ios/file/cloud_done/materialiconsround/black/round_cloud_done_black_48pt.xcassets/round_cloud_done_black_48pt.imageset/round_cloud_done_black_48pt_2x.png

Since 3.1: Under the "General" settings, you can click on "Create Debug Archive ..." to pick the location of where the desktop client will export the logs and the database to a zip file.
On previous releases: Via the command line: `nextcloud --logdebug --logwindow` or `nextcloud --logdebug --logfile log.txt`
(See also https://docs.nextcloud.com/desktop/3.0/troubleshooting.html#log-files)

2. Web server error log:

3. Server logfile: nextcloud log (data/nextcloud.log):

**Comment by er-vin** [comment]
I think we have other users with that many files, it's slow probably but not as dramatic as this + the memory consumption.

I wonder what is at play there. When you say 200 000 that's counting the excluded files which are apparently not excluded?

**Comment by HorstBort** [comment]
Yes, that's including excluded files. I also just noticed, that the client always checks for changes in unsynced subfolders. I'll create another issue for that.

Also: I could not find a way to reduce the log level in the client - when it's unresponsive, it is compressing the multi GB log files, apparently. In everyday use, all those debug and info entries are not necessary, I'd presume.

**Comment by HorstBort** [comment]
Probably related: #2855

**Comment by er-vin** [comment]
> Also: I could not find a way to reduce the log level in the client - when it's unresponsive, it is compressing the multi GB log files, apparently. In everyday use, all those debug and info entries are not necessary, I'd presume.

Right, with the change in defaults --logdebug is kind of useless in 3.1. What you want to do is put `logDebug=false` in the `General` group of your nextcloud.cfg

**Comment by er-vin** [comment]
Now I still wonder why it's multi GB log files still. In our tests before flipping that default even for large amount of files it didn't get that bad I think. I wonder if there's something else than sheer amount of files which makes it log like crazy in your case.

Try to pass --logfile on the command line, it should then stop using the default log dir and won't compress. Might be an easier way to find out what it's spammed with apart from the instruction new.

**Comment by HorstBort** [comment]
I tried with passing `--logfile /dev/null`. Did not help that much, syncs are still slow, as far as I can see b/c of #2855.

As a workaround, I have changed my folder layout and used individual folder sync connections for the folders I want synced instead of excluding subfolder, i.e. instead of 

    - [/] Documents
      - [X] Folder A
      - [ ] Folder B (not synced, containing a few hundred thousand files)
      - [X] Folder C
 individual connections for `Folder A` and `Folder C`. Now performance is good.

**Comment by er-vin** [comment]
> I tried with passing `--logfile /dev/null`. Did not help that much, syncs are still slow, as far as I can see b/c of #2855.

Right but still you don't get the high CPU/RAM usage and unresponsiveness anymore you reported initially since it's not compressing those logs, right?

I'd still be curious about what it spits in those logs that they are so big for your case... I wouldn't expect it to choke on that amount of files I think.

**Comment by grka** [comment]
Hi there,

I'm facing a similar issue on under Mac OS Big.
In previous OS version (Catalina) the client was working well. 

Now : 
-> It generates a lot of log files (I have tried `logDebug=false` in nextcloug.cfg but it continues to output a lot of logs). 
-> The Mac OS monitoring system tells that the client have a huge energy impact (my battery comes from 100% to 0% in about one hour with only the client running). 
-> The log files tells that the client detected changes in all my folders and subfolders (and in fact there is no change). So the client is trying to find difference between my local and distant files for all files (as far I understand the log file). 

And, weird : 
-> When after I check and uncheck the checkbox to sync hidden files the client goes back to a low energy consuption and does not log a lot anymore for about 30mn and then consume a lot. 

I tryed : 
-> updating the client to the latest version
-> updating the server to the latest 18x stable version
-> deleteting the client, the synced folder and reinstalling it. 

But no matter what I do it stills run to a huge log output and high energy impact. 

So I'm a bit frustrated and I don't know how I can figure out this !

**Comment by er-vin** [comment]
> Hi there,
> 
> I'm facing a similar issue on under Mac OS Big.
> In previous OS version (Catalina) the client was working well.
> 
> Now :
> -> It generates a lot of log files (I have tried `logDebug=false` in nextcloug.cfg but it continues to output a lot of logs).
> -> The Mac OS monitoring system tells that the client have a huge energy impact (my battery comes from 100% to 0% in about one hour with only the client running).
> -> The log files tells that the client detected changes in all my folders and subfolders (and in fact there is no change). So the client is trying to find difference between my local and distant files for all files (as far I understand the log file).
> 
> And, weird :
> -> When after I check and uncheck the checkbox to sync hidden files the client goes back to a low energy consuption and does not log a lot anymore for about 30mn and then consume a lot.
> 
> I tryed :
> -> updating the client to the latest version
> -> updating the server to the latest 18x stable version
> -> deleteting the client, the synced folder and reinstalling it.
> 
> But no matter what I do it stills run to a huge log output and high energy impact.
> 
> So I'm a bit frustrated and I don't know how I can figure out this !

For sure the logs aren't really the cause but a symptom in any case. So I guess the real problem is the fact that changes got wrongly detected for some reason.

Something happened server side before it did this? Like a wrong restore which changed all the etag or any similar sysadmin mistake?

**Comment by grka** [comment]
I didn't touched the server since a long time before this issue (I manage it)
I will make some more checks. 

I agree on the fact that the log output must be a symptom not a cause. 
But it's quite weird that it recheck all the files because there is no "change" apparently (the files timespamps haren't event been "touched")

---> I suspect it's linked to the Mac OS Big update and that's related to the Apple "Time Machine" tool that backup my drive. I stopped the Time Machines crons and it seems that the nextcloud client is not having the issue any more. 
---> I will relaunch a tiime machine manually and look at the nextcloud behaviour and lease a reply here to inform

**Comment by grka** [comment]
Just to inform that it doesn't seem to be related to time machine. 
After switching the checkbox after random time it goes back to 100% cpu usage ...

**Comment by cybarspace** [comment]
My issue is similar but doesn't involve a log file.
For a few days (less than a week), the Nextcloud client app suddenly occupies 100% RAM and makes the system freeze such that I have to hold the power button to shutdown. Sorry for not being able to provide logs, I don't know how to generate them.

My system: Manjaro XFCE + i3, Linux 5.10.15, i5-5200U, 8GB RAM
Nextcloud client: version 3.1.2
Nextcloud server:
```
Name       Version      Rev    Tracking       Publisher   Notes
nextcloud  20.0.7snap1  26119  latest/stable  nextcloud✓  -
```
Server info:
```
snap    2.48.3+20.04
snapd   2.48.3+20.04
series  16
ubuntu  20.04
kernel  5.4.0-65-generic
```

**Comment by er-vin** [comment]
> Sorry for not being able to provide logs, I don't know how to generate them.

They should be in ~/.config/Nextcloud/logs

**Comment by cybarspace** [comment]
> They should be in ~/.config/Nextcloud/logs

Oh, thanks. There's only one log file but it's 578 MB. And it's too many lines to paste here. Should I look for any specific part in the log for anything weird that may be going on?

Also, this just happened:
![image](https://user-images.githubusercontent.com/50134239/107874393-a59b9300-6ee3-11eb-9b03-296724113ed1.png)
(also attached to my previous comment for completeness)

Thankfully I noticed the RAM ramping up moments before my system would freeze and terminated the process, but it'd be nice if I didn't have to look out for it in constant terror... :sweat_smile:

**Comment by er-vin** [comment]
Yeah seems to be like the other ones for whom suddenly the logs get too large and then compression on rotate makes the memory blow up... I wonder why it gets so large in the first place though. I guess the strategy for the rotation should be adjusted to make sure it's not trying to compress too large payloads.

**Comment by cybarspace** [comment]
Meanwhile, is there an option to turn off logs until this issue is fixed, or to limit the size of logs? Anything to get the PC to stop freezing would be much appreciated. Thanks. :sweat_smile:

**Comment by Quentin59000** [comment]
As for me I rolled back to 3.0.3 client version. 
No huge logs being generated anymore and the system came back stable

**Comment by HorstBort** [comment]
Might have something to do with debug messages appearing in log files even though nextcloud is started **without** `--logdebug` as mentioned in my original report:

    for II in 20210216_1*_owncloud.log*; do
    echo "###" file:$II; echo "    containing debug": $(zgrep debug $II | wc -l)
    echo not containing debug: $(zgrep -v debug $II | wc -l); echo
    done
    ### file:20210216_1249_owncloud.log.0.gz
        containing debug: 15159
    not containing debug: 8312
    
    ### file:20210216_1318_owncloud.log.0.gz
        containing debug: 1058
    not containing debug: 2108
    
    ### file:20210216_1339_owncloud.log.0.gz
        containing debug: 624
    not containing debug: 330
    
    ### file:20210216_1339_owncloud.log.1.gz
        containing debug: 5741
    not containing debug: 83
    
    ### file:20210216_1339_owncloud.log.2.gz
        containing debug: 20826
    not containing debug: 9058
    
    ### file:20210216_1339_owncloud.log.3.gz
        containing debug: 9340
    not containing debug: 1370
    
    ### file:20210216_1348_owncloud.log.0.gz
        containing debug: 9215
    not containing debug: 896
    
    ### file:20210216_1350_owncloud.log.0.gz
        containing debug: 8875
    not containing debug: 635
    
    ### file:20210216_1351_owncloud.log.0.gz
        containing debug: 9043
    not containing debug: 700
    
    ### file:20210216_1351_owncloud.log.1.gz
        containing debug: 9126
    not containing debug: 695
    
    ### file:20210216_1351_owncloud.log.2.gz
        containing debug: 8861
    not containing debug: 612
    
    ### file:20210216_1352_owncloud.log.0.gz
        containing debug: 6158
    not containing debug: 652
    
    ### file:20210216_1358_owncloud.log.0.gz
        containing debug: 5893
    not containing debug: 223
    
    ### file:20210216_1358_owncloud.log.1.gz
        containing debug: 6131
    not containing debug: 616
    
    ### file:20210216_1404_owncloud.log.0.gz
        containing debug: 5864
    not containing debug: 211
    
    ### file:20210216_1404_owncloud.log.1
        containing debug: 665
    not containing debug: 1487

**Comment by ElStellino** [comment]
I am no developer, just an end user, and today after synchronizing a folder containing videos in one of the clients, Nextcloud said there was no space on the server (which is strange, as the server has a dedicated 2TB HDD). 
Coming back to this Linux Mint 20.1 Cinnamon machine to check, I saw there was a warning that popped up, saying insufficient space in the /system folder.
As the space was almost finishing, I didn’t have time to check what was added in /system, also because with 8MB left it was getting rather unresponsive. I then deleted all the files on the server, emptied the bin, and boom, 34.9 GB free (it’s a 60GB partition)
I barely managed to see that /var/log was rather big.
Thanks!
